#Import the Data

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
attorney = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/attorneys.csv")
attTimeEntires = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/attorneytimeentries.csv")
cat = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/categories.csv")
client = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/clients.csv")
questionP = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/questionposts.csv")
questions = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/questions.csv")
statesite = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/statesites.csv")
subcat = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/subcategories.csv")

<ipython-input-247-6befc2724c31>:5: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  questionP = pd.read_csv("/content/drive/Shareddrives/Data Fest 2023/data/questionposts.csv")


In [ ]:
attorney.head()

Id StateAbbr                           AttorneyUno          City  \
0   1        NH  7A2C12BE-9F37-4C9D-A6ED-41B916820CE0       Concord   
1   2        NH  29A49AA6-047E-4008-ABBD-3DE57156B526       Concord   
2   3        MS  9DF11B65-7F29-452C-A69E-54CD72E4A116       Jackson   
3   4        VA  693FD55F-5F58-405A-B389-5E78B3406497      Richmond   
4   5        CT  0AC2A866-19CD-4D1A-94B7-BF41A1430553  Wethersfield   

            County      StateName  PostalCode           CreatedUtc  
0         Cheshire  New Hampshire        3301  2019-02-14 16:21:52  
1        Merrimack  New Hampshire        3229  2019-02-14 18:14:30  
2           Rankin    Mississippi       39203  2016-08-19 17:07:23  
3  Richmond (city)       Virginia       23219  2016-08-19 18:22:53  
4         Hartford    Connecticut        6410  2016-08-21 01:21:13

In [ ]:
attTimeEntires.head()

Id StateAbbr                          TimeEntryUno  \
0   1        AK  44954D1D-C354-41F6-8C33-327E866B6BA3   
1   2        AL  F1D8F733-1BDB-47FD-BC7C-CB512E2C5A3A   
2   3        AR  81ED2628-2FF4-4B67-86B1-F9DFEC603C15   
3   4        AZ  14B1BCC4-CEF6-4108-AB54-3CDE482845F0   
4   5        CA  41275A72-CD69-4A42-85FB-E3E4FEE0C316   

                            AttorneyUno  Hours         EnteredOnUtc  
0  C126C566-9039-4991-BDD8-0CDD05C27DDF    1.8  2017-01-05 21:25:23  
1  F8C0E3D3-B3A5-42B9-9E2F-2A637BA83D6E    0.4  2020-03-24 01:10:56  
2  34BFCC3D-B107-4164-B405-DEFF63A65B2D    0.4  2016-12-02 23:21:46  
3  875DB0F4-619B-4A13-8104-85A3C15D90C4    0.8  2015-12-16 20:32:55  
4  22746734-D076-4AC8-BA66-50426C01093C    3.0  2020-09-04 00:02:54

In [ ]:
cat.head()

Id StateAbbr                           CategoryUno  \
0   1        AK  4CB7A73E-AAB0-4E8F-94D1-8AC8610AAC64   
1   2        AL  08C7C59B-6AAE-4BC1-83FE-BC85F1544260   
2   3        AR  FD5059F7-DE7A-4929-8D67-E72C80773BAD   
3   4        AZ  D90F1BC7-CD04-409A-9AF7-D4DAB649425A   
4   5        CA  FD678F86-62B1-4EC4-A628-98AEE7BC74D5   

                       Category  
0  Consumer Financial Questions  
1  Consumer Financial Questions  
2  Consumer Financial Questions  
3  Consumer Financial Questions  
4  Consumer Financial Questions

In [ ]:
client.head()

Id StateAbbr                             ClientUno      County  \
0   1        MS  B731DA7A-A97A-4A2E-BF11-0A126A05AF70  Washington   
1   2        MS  8101DB7C-0CC8-4DE9-9E87-71137269AAD5    Harrison   
2   3        GA  DD5F1EFC-8298-457F-BC09-CABBEB57A697      Barrow   
3   4        TX  21DC9E3D-D0AA-4B85-88BB-428F5BCF7F13     Lubbock   
4   5        ID  3782D4F9-9B16-4BC8-ADA3-F85CEE200630       Boise   

     StateName  PostalCode      EthnicIdentity   Age  Gender  \
0  Mississippi     73179.0                 NaN   NaN     NaN   
1  Mississippi     39560.0           Caucasian   NaN    Male   
2      Georgia     30157.0  Latino or Hispanic  55.0    Male   
3        Texas     79416.0               Other  61.0  Female   
4        Idaho     83705.0           Caucasian  65.0    Male   

         MaritalStatus Veteran Imprisoned  NumberInHousehold  AnnualIncome  \
0                  NaN     NaN         No                3.0       26000.0   
1  Divorced or Widowed     NaN         No                2.0          52.0   
2               Single     NaN         No                5.0        3000.0   
3  Married / remarried      No         No                4.0       15600.0   
4               Single      No         No                1.0       15000.0   

   AllowedIncome  CheckingBalance  SavingsBalance  InvestmentsBalance  \
0        37190.0            300.0             NaN                 NaN   
1        25390.0              1.0             1.0                 NaN   
2        60790.0              NaN             NaN                 NaN   
3        48990.0            100.0             NaN                 NaN   
4        13590.0            580.0            25.0                 NaN   

            CreatedUtc  
0  2016-08-25 15:15:37  
1  2016-08-26 00:55:17  
2  2016-10-12 14:37:06  
3  2017-01-01 19:57:07  
4  2017-02-23 00:03:27

In [ ]:
questionP.head()

Id StateAbbr                           QuestionUno  \
0  1        AK  B7C95D50-6C81-4E61-BC16-81077B7D3F7E   
1  2        AK  C579CE49-6B7A-44E7-A203-5D3E52623804   
2  3        AK  F3A78AD3-D3D4-447F-BEAF-F62F1D1D4CCE   
3  4        AK  5F8506D3-D9F1-48EE-8081-D30259DBF4E8   
4  5        AK  B4C814C9-8983-413F-8BC4-F2BE01665555   

                                            PostText     CreatedUtc  
0  Do you still need help with this issue?  It ha...  4/25/18 18:09  
1                                           Anyone?,  4/17/18 21:15  
2  Yes, those are the right forms.  I'm sorry, I ...  4/18/18 17:51  
3  Hi ###,You do make some good points.  The diff...  4/18/18 22:15  
4  My father passed away and I have DNA test resu...  4/12/18 10:52

In [ ]:
questions.head()

Id StateAbbr                           QuestionUno  \
0   1        NC  BB53B2A4-7C1B-45C3-9693-62047889317D   
1   2        NC  0C49F841-2782-4313-961E-7882B96D0854   
2   3        NC  BFC28E08-B6B1-4185-A91A-89DF90816BE4   
3   4        NC  45B24109-243A-489B-B0D2-2F36C2A369FE   
4   5        NC  C4707F71-FE4A-4154-8174-AF00BEE2B361   

                            CategoryUno                      Category  \
0  C932D197-7F43-4BCF-BE0B-FE5F5DBCB9EF           Family and Children   
1  C8E1275A-6C42-42E4-9B7D-231BAC438FC9             Individual Rights   
2  C8E1275A-6C42-42E4-9B7D-231BAC438FC9             Individual Rights   
3  C932D197-7F43-4BCF-BE0B-FE5F5DBCB9EF           Family and Children   
4  F6016FBD-9501-4448-BCD3-BBB0E834E33B  Consumer Financial Questions   

                         SubcategoryUno  \
0  E9F87919-0CD5-4D9D-AAC6-3CBC49132F1D   
1  73A59926-BA0B-44EC-ABF3-F4C9181EEA83   
2  73A59926-BA0B-44EC-ABF3-F4C9181EEA83   
3  E9F87919-0CD5-4D9D-AAC6-3CBC49132F1D   
4  73FE486E-2B51-456A-A6D9-78A7D05802AA   

                                       Subcategory  \
0  Family/Divorce/Custody/Visitation/Child Support   
1                      Civil/Constitutional Rights   
2                      Civil/Constitutional Rights   
3  Family/Divorce/Custody/Visitation/Child Support   
4                    Bankruptcy, Debts & Purchases   

                       AskedByClientUno           AskedOnUtc  \
0  9C35423C-914A-42BC-8EBC-A370E76E2178  2019-06-19 22:10:06   
1  DF636AEE-7D08-4F90-9A5C-0A2A726C6B59  2019-06-19 23:34:36   
2  DF636AEE-7D08-4F90-9A5C-0A2A726C6B59  2019-06-19 23:57:18   
3  C9C3FA40-0D60-49B9-97F1-63AFEFAFA484  2019-06-20 02:33:14   
4  ED7DEA10-CDB7-4A05-891D-889D1EA710BE  2019-06-20 17:04:47   

                     TakenByAttorneyUno           TakenOnUtc  \
0                                   NaN                  NaN   
1                                   NaN                  NaN   
2  FA34142B-1575-4720-981C-2D28C3560137  2019-06-22 02:37:16   
3  FA34142B-1575-4720-981C-2D28C3560137  2019-07-27 18:04:08   
4                                   NaN                  NaN   

                    ClosedByAttorneyUno          ClosedOnUtc  \
0                                   NaN  2019-08-13 15:09:06   
1                                   NaN  2019-08-13 15:09:14   
2  FA34142B-1575-4720-981C-2D28C3560137  2019-06-24 19:02:35   
3  FA34142B-1575-4720-981C-2D28C3560137  2019-07-31 05:49:18   
4                                   NaN  2019-08-13 15:09:20   

         LegalDeadline  
0  2019-07-08 00:00:00  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN

In [ ]:
statesite.head()

Id StateAbbr    StateName  AllowedAssets  BaseIncomeLimit  \
0   1        AL      Alabama          10000            13590   
1   2        AR     Arkansas          10000            13590   
2   3        AZ      Arizona          10000            13590   
3   4        CA   California          10000            13590   
4   5        CT  Connecticut          10000            13590   

   PerHouseholdMemberIncomeLimit  IncomeMultiplier  
0                           4720               2.5  
1                           4720               2.5  
2                           4720               2.5  
3                           4720               4.0  
4                           4720               2.5

In [ ]:
subcat.head()

Id StateAbbr                           CategoryUno  \
0   1        AK  2D43B4FD-FAC8-47C3-A891-085F5D74D64B   
1   2        AK  2D43B4FD-FAC8-47C3-A891-085F5D74D64B   
2   3        AK  2D43B4FD-FAC8-47C3-A891-085F5D74D64B   
3   4        AK  D3504F18-AC99-408B-B1D2-230DC130F9EC   
4   5        AK  BF144F3C-5743-47C5-9EEB-418FF0FAF0D8   

                         SubcategoryUno                   Subcategory  
0  546C6AD6-4D2B-4B1D-B09F-8016C5B02D66              COVID-19-Housing  
1  572148E8-96C7-4B74-8280-B749191C5B4C            COVID-19-Insurance  
2  5AA5088E-3A9D-4D99-B6FE-69122C34194A       COVID-19-Lost Documents  
3  E5DF5CEB-649F-47B3-B665-1BEB3EA4AD42  Emancipation and Delinquency  
4  16E4350E-BC59-40B7-BF72-0BC225ECA4D1                   Immigration

#Extract Unanswered Questions



In [ ]:
nottaken = questions[questions['TakenOnUtc'].isna()]
#nottaken.shape
#questions.shape
nottaken.shape[0] /questions.shape[0] * 100

28.20794660856964

There were 57228 out of 202879 questions not taken by the attorneys. That is 28.2 percent of the questions.

In [ ]:
#nottaken

In [ ]:
grouped = nottaken.groupby(['Category', 'Subcategory'])

grouped.first()

Id  \
Category                          Subcategory                                 
Consumer Financial Questions      Auto or Property Insurance           1993   
                                  Bankruptcy                          11796   
                                  Bankruptcy                          49855   
                                  Bankruptcy or Debtor Relief Cases  124804   
                                  Bankruptcy, Debts & Purchases           5   
...                                                                     ...   
Work, Employment and Unemployment Work, Employment, or Unemployment  150035   
                                  Work/Employment                    186749   
                                  Work/Employment/Unemployment           29   
                                  Work/Unemployment                  190720   
                                  Worker's Comp                      119009   

                                                                    StateAbbr  \
Category                          Subcategory                                   
Consumer Financial Questions      Auto or Property Insurance               MO   
                                  Bankruptcy                               TX   
                                  Bankruptcy                               NY   
                                  Bankruptcy or Debtor Relief Cases        IN   
                                  Bankruptcy, Debts & Purchases            NC   
...                                                                       ...   
Work, Employment and Unemployment Work, Employment, or Unemployment        NM   
                                  Work/Employment                          MS   
                                  Work/Employment/Unemployment             NC   
                                  Work/Unemployment                        CT   
                                  Worker's Comp                            AK   

                                                                                              QuestionUno  \
Category                          Subcategory                                                               
Consumer Financial Questions      Auto or Property Insurance         51324C6D-7073-4A73-8C2A-FB5B57A93FF5   
                                  Bankruptcy                         6576AEFB-603B-43DE-86C7-95A0D7DBDEBB   
                                  Bankruptcy                         B385D21A-D9A9-4CE5-B646-7EC94F00232E   
                                  Bankruptcy or Debtor Relief Cases  A62CB07A-DFED-41A5-A5A9-98B352C0185D   
                                  Bankruptcy, Debts & Purchases      C4707F71-FE4A-4154-8174-AF00BEE2B361   
...                                                                                                   ...   
Work, Employment and Unemployment Work, Employment, or Unemployment  10BA589E-438F-4430-B6BA-5EFB5DEF8E1E   
                                  Work/Employment                    5DC557C4-F811-4FEF-8B01-05483BDB4C80   
                                  Work/Employment/Unemployment       309F5951-B58B-4DA4-A518-B04BCEA409FF   
                                  Work/Unemployment                  90FD4022-65C2-409E-B61B-239ACC5BDE3C   
                                  Worker's Comp                      00008AE7-5B06-454D-BA0F-969CE8809C33   

                                                                                              CategoryUno  \
Category                          Subcategory                                                               
Consumer Financial Questions      Auto or Property Insurance         1EBBF19C-9E26-408A-805A-BC5362E36F28   
                                  Bankruptcy                         FA45ADC5-94B4-4597-B167-E715EC0E47E2   
                                  Bankruptcy                         0400294A-D08E-437F-8577-A7A0B9D254A0   
                            

#Preprocessing Question Data

In [ ]:
ques = questions.drop(columns=['Id', 'CategoryUno','SubcategoryUno','AskedByClientUno','AskedOnUtc' ,  'ClosedByAttorneyUno', 'ClosedOnUtc' ])
ques

StateAbbr                           QuestionUno  \
0             NC  BB53B2A4-7C1B-45C3-9693-62047889317D   
1             NC  0C49F841-2782-4313-961E-7882B96D0854   
2             NC  BFC28E08-B6B1-4185-A91A-89DF90816BE4   
3             NC  45B24109-243A-489B-B0D2-2F36C2A369FE   
4             NC  C4707F71-FE4A-4154-8174-AF00BEE2B361   
...          ...                                   ...   
202874        NC  376FFF48-9068-4A8F-9316-2EB5A7E38001   
202875        NC  98AA49B1-D423-4D12-9E70-A5BD6663DFE5   
202876        NC  B80305AA-7509-436E-AFF8-6B33AC682730   
202877        NC  DC374484-7C6E-44A3-9DB7-89B8868518F1   
202878        NC  1F2F2FC1-4661-4A14-A7BE-43E883D1BD44   

                            Category  \
0                Family and Children   
1                  Individual Rights   
2                  Individual Rights   
3                Family and Children   
4       Consumer Financial Questions   
...                              ...   
202874                         Other   
202875                         Other   
202876                         Other   
202877                         Other   
202878           Family and Children   

                                            Subcategory  \
0       Family/Divorce/Custody/Visitation/Child Support   
1                           Civil/Constitutional Rights   
2                           Civil/Constitutional Rights   
3       Family/Divorce/Custody/Visitation/Child Support   
4                         Bankruptcy, Debts & Purchases   
...                                                 ...   
202874                                  Civil Procedure   
202875                                  Civil Procedure   
202876                                  Civil Procedure   
202877                                            Other   
202878  Family/Divorce/Custody/Visitation/Child Support   

                          TakenByAttorneyUno           TakenOnUtc  \
0                                        NaN                  NaN   
1                                        NaN                  NaN   
2       FA34142B-1575-4720-981C-2D28C3560137  2019-06-22 02:37:16   
3       FA34142B-1575-4720-981C-2D28C3560137  2019-07-27 18:04:08   
4                                        NaN                  NaN   
...                                      ...                  ...   
202874  6E43EC98-8700-4439-98A9-52B663CCE348  2019-07-17 19:22:29   
202875  FA34142B-1575-4720-981C-2D28C3560137  2019-06-22 03:07:59   
202876                                   NaN                  NaN   
202877  6E43EC98-8700-4439-98A9-52B663CCE348  2019-06-24 14:47:57   
202878                                   NaN                  NaN   

              LegalDeadline  
0       2019-07-08 00:00:00  
1                       NaN  
2                       NaN  
3                       NaN  
4                       NaN  
...                     ...  
202874                  NaN  
202875  2019-06-18 00:00:00  
202876  2019-06-27 00:00:00  
202877  2019-07-01 00:00:00  
202878                  NaN  

[202879 rows x 7 columns]

In [ ]:
questionP

Id StateAbbr                           QuestionUno  \
0            1        AK  B7C95D50-6C81-4E61-BC16-81077B7D3F7E   
1            2        AK  C579CE49-6B7A-44E7-A203-5D3E52623804   
2            3        AK  F3A78AD3-D3D4-447F-BEAF-F62F1D1D4CCE   
3            4        AK  5F8506D3-D9F1-48EE-8081-D30259DBF4E8   
4            5        AK  B4C814C9-8983-413F-8BC4-F2BE01665555   
...        ...       ...                                   ...   
356767  356954        UT  D1CD238C-4216-4F3D-BFB4-3C8D28C3B8C6   
356768  356955        UT  D1CD238C-4216-4F3D-BFB4-3C8D28C3B8C6   
356769  356956        UT  D1CD238C-4216-4F3D-BFB4-3C8D28C3B8C6   
356770  356957        UT  B9A70FE8-71CA-42F2-B7D4-8DE07E7DD41D   
356771  356958        UT  B9A70FE8-71CA-42F2-B7D4-8DE07E7DD41D   

                                                 PostText     CreatedUtc  
0       Do you still need help with this issue?  It ha...  4/25/18 18:09  
1                                                Anyone?,  4/17/18 21:15  
2       Yes, those are the right forms.  I'm sorry, I ...  4/18/18 17:51  
3       Hi ###,You do make some good points.  The diff...  4/18/18 22:15  
4       My father passed away and I have DNA test resu...  4/12/18 10:52  
...                                                   ...            ...  
356767  to ensure that the question is clear and not w...  4/25/19 12:44  
356768  Looking at your decree, and just generally as ...  4/25/19 16:29  
356769  Thank you again for your time. Thus far no one...  4/30/19 13:56  
356770  My neighbors behind me have ###. He recently c...  4/23/19 22:32  
356771  Hi ###,Thank you for submitting your question....   5/9/19 23:11  

[356772 rows x 5 columns]

In [ ]:
#qgroup = questionP.groupby('QuestionUno', group_keys=True).apply(lambda x: x)
qPid = questionP['QuestionUno']
qPid.drop_duplicates()

uniqueques = qPid.drop_duplicates()
uniqueques.index

Int64Index([     0,      1,      2,      3,      4,      5,      8,      9,
                10,     11,
            ...
            356744, 356746, 356748, 356750, 356752, 356755, 356757, 356762,
            356764, 356770],
           dtype='int64', length=148238)

In [ ]:
reducedQ = questionP.iloc[uniqueques.index]
reducedQ


Id StateAbbr                           QuestionUno  \
0            1        AK  B7C95D50-6C81-4E61-BC16-81077B7D3F7E   
1            2        AK  C579CE49-6B7A-44E7-A203-5D3E52623804   
2            3        AK  F3A78AD3-D3D4-447F-BEAF-F62F1D1D4CCE   
3            4        AK  5F8506D3-D9F1-48EE-8081-D30259DBF4E8   
4            5        AK  B4C814C9-8983-413F-8BC4-F2BE01665555   
...        ...       ...                                   ...   
356755  356942        UT  DC4952A1-2508-4ABD-9C23-80C6362FCB87   
356757  356944        UT  A9C002F9-BFCD-4E6B-AC4A-EA60357DBDA1   
356762  356949        UT  B1E8E1D9-12A0-477C-869F-93792E6098D0   
356764  356951        UT  D1CD238C-4216-4F3D-BFB4-3C8D28C3B8C6   
356770  356957        UT  B9A70FE8-71CA-42F2-B7D4-8DE07E7DD41D   

                                                 PostText           CreatedUtc  
0       Do you still need help with this issue?  It ha...        4/25/18 18:09  
1                                                Anyone?,        4/17/18 21:15  
2       Yes, those are the right forms.  I'm sorry, I ...        4/18/18 17:51  
3       Hi ###,You do make some good points.  The diff...        4/18/18 22:15  
4       My father passed away and I have DNA test resu...        4/12/18 10:52  
...                                                   ...                  ...  
356755  I made ###. The first order arrived and I assu...  019-04-20 04:23:23"  
356757  I am filing for Chapter ###.  I live with my M...        4/22/19 18:56  
356762  To Whom It May Concern,My deceased father was ...        4/22/19 21:05  
356764  Divorce decree went into affect ######6. Respo...  019-04-22 23:42:03"  
356770  My neighbors behind me have ###. He recently c...        4/23/19 22:32  

[148238 rows x 5 columns]

In [ ]:
#reducedQ.to_csv('/content/drive/Shareddrives/Data Fest 2023/data/reducedQuestions', index=False)

In [ ]:
mergedq = pd.merge(reducedQ, questions, on='QuestionUno')
mergedq = mergedq.drop(columns = ['Id_x', 'StateAbbr_x', 'CreatedUtc','Id_y', 'AskedOnUtc',  'ClosedByAttorneyUno', 'ClosedOnUtc'])

mergedq = mergedq.rename(columns={"StateAbbr_y": "StateAbbr"})
mergedq

QuestionUno  \
0       B7C95D50-6C81-4E61-BC16-81077B7D3F7E   
1       C579CE49-6B7A-44E7-A203-5D3E52623804   
2       F3A78AD3-D3D4-447F-BEAF-F62F1D1D4CCE   
3       5F8506D3-D9F1-48EE-8081-D30259DBF4E8   
4       B4C814C9-8983-413F-8BC4-F2BE01665555   
...                                      ...   
148203  DC4952A1-2508-4ABD-9C23-80C6362FCB87   
148204  A9C002F9-BFCD-4E6B-AC4A-EA60357DBDA1   
148205  B1E8E1D9-12A0-477C-869F-93792E6098D0   
148206  D1CD238C-4216-4F3D-BFB4-3C8D28C3B8C6   
148207  B9A70FE8-71CA-42F2-B7D4-8DE07E7DD41D   

                                                 PostText StateAbbr  \
0       Do you still need help with this issue?  It ha...        AK   
1                                                Anyone?,        AK   
2       Yes, those are the right forms.  I'm sorry, I ...        AK   
3       Hi ###,You do make some good points.  The diff...        AK   
4       My father passed away and I have DNA test resu...        AK   
...                                                   ...       ...   
148203  I made ###. The first order arrived and I assu...        UT   
148204  I am filing for Chapter ###.  I live with my M...        UT   
148205  To Whom It May Concern,My deceased father was ...        UT   
148206  Divorce decree went into affect ######6. Respo...        UT   
148207  My neighbors behind me have ###. He recently c...        UT   

                                 CategoryUno                      Category  \
0       2D43B4FD-FAC8-47C3-A891-085F5D74D64B      Housing and Homelessness   
1       2D43B4FD-FAC8-47C3-A891-085F5D74D64B      Housing and Homelessness   
2       A96DEADE-C287-4B31-BFF5-6C02E0640001           Family and Children   
3       E31B199A-B01E-47E6-86DB-8D99B2BB7420                         Other   
4       A96DEADE-C287-4B31-BFF5-6C02E0640001           Family and Children   
...                                      ...                           ...   
148203  B4FA50DD-8B47-4BD9-8F78-91E2AF96A31D                         Other   
148204  C955358F-23B0-4BCF-948D-C7AE744536DE  Consumer Financial Questions   
148205  B4FA50DD-8B47-4BD9-8F78-91E2AF96A31D                         Other   
148206  39A338D3-FE08-4A0B-90FA-CEC4DBB2C64F           Family and Children   
148207  55739525-2665-43A6-9242-6A036EE81E55      Housing and Homelessness   

                              SubcategoryUno                Subcategory  \
0       6F28DF84-2BDA-48AE-97F8-A083D20E3EC8  Housing or Property Owned   
1       6F28DF84-2BDA-48AE-97F8-A083D20E3EC8  Housing or Property Owned   
2       372E3CF8-3543-4C71-BA48-DA3F52FC5792     Family/Divorce/Custody   
3       3DBF3B9F-9ED0-4028-A4FE-0823B0012634                      Other   
4       BB9683D7-FAE6-4E33-AF74-52E71373C8D7          Wills/Inheritance   
...                                      ...                        ...   
148203  290A8D28-2BB6-499F-ACD2-4C3F284587D2                      Other   
148204  853609D4-68B8-4C13-A9CD-50562473A304          Debts & Purchases   
148205  290A8D28-2BB6-499F-ACD2-4C3F284587D2                      Other   
148206  178DFF3C-856C-4100-B819-9659C3ECFC4E     Family/Divorce/Custody   
148207  796DFA43-55B0-48FA-8204-7B00135305DC  Housing or Property Owned   

                            AskedByClientUno  \
0       56A7173F-C8B2-4E17-8189-DBDB8EDDAA1C   
1       6CB2B648-61A3-425F-9B84-A7C637B83126   
2       00BB08E8-38C6-42CE-8A01-6C8A1F7F88E7   
3       6CB2B648-61A3-425F-9B84-A7C637B83126   
4       2761B481-9CAD-4494-B8CB-27AF221FDA69   
...                                      ...   
148203  3B8208D2-8246-48EC-8531-03C5150DC87E   
148204  4F7851EF-7C2C-4DD6-B1C6-1EA8A37D0735   
148205  1A6DFAFF-3DE5-4E38-92B6-C6F24C69FDA8   
148206  2C9C84D9-4329-4FD7-9877-6491EFB2DE83   
148207  B75DF747-F207-499E-A2BA-AE95751D95B3   

                          TakenByAttorneyUno           TakenOnUtc  \
0       0293D737-A73B-47D8-8D26-FE5AB33F67E1  2018-04-25 18:08:21   
1                                        NaN      

In [ ]:
#mergedq.to_csv('/content/drive/Shareddrives/Data Fest 2023/data/MergedQuestions.csv', index=False)

#Attorney Info

In [ ]:
attorney

Id StateAbbr                           AttorneyUno            City  \
0          1        NH  7A2C12BE-9F37-4C9D-A6ED-41B916820CE0         Concord   
1          2        NH  29A49AA6-047E-4008-ABBD-3DE57156B526         Concord   
2          3        MS  9DF11B65-7F29-452C-A69E-54CD72E4A116         Jackson   
3          4        VA  693FD55F-5F58-405A-B389-5E78B3406497        Richmond   
4          5        CT  0AC2A866-19CD-4D1A-94B7-BF41A1430553    Wethersfield   
...      ...       ...                                   ...             ...   
11539  11540        VA  F2EC8DAD-D7B0-40B8-AFF2-F5705E2F59B0      Washington   
11540  11541        VA  386CD203-A6E0-4C8F-90D3-0EEEC2701CA3  Virginia Beach   
11541  11542        VA  1C366F1B-0990-4847-8B47-D56D426AD964  Virginia Beach   
11542  11543        VA  FE8B3662-7C55-416A-B123-C3F9A9F3CAFA        Richmond   
11543  11544        VA  164F3C9B-FA80-427B-8172-8A4ECE7DAC24        Richmond   

                    County      StateName  PostalCode           CreatedUtc  
0                 Cheshire  New Hampshire        3301  2019-02-14 16:21:52  
1                Merrimack  New Hampshire        3229  2019-02-14 18:14:30  
2                   Rankin    Mississippi       39203  2016-08-19 17:07:23  
3          Richmond (city)       Virginia       23219  2016-08-19 18:22:53  
4                 Hartford    Connecticut        6410  2016-08-21 01:21:13  
...                    ...            ...         ...                  ...  
11539            Arlington       Virginia       20037  2021-11-03 15:16:31  
11540  Virginia Beach city       Virginia       23451  2021-11-04 13:56:07  
11541  Virginia Beach city       Virginia       23451  2021-11-15 20:31:02  
11542        Richmond city       Virginia       23225  2021-11-15 23:27:59  
11543        Richmond city       Virginia       23227  2021-11-17 19:26:07  

[11544 rows x 8 columns]

In [ ]:
newq = questions.dropna(subset=['ClosedByAttorneyUno'])
newq = newq.rename(columns={"ClosedByAttorneyUno": "AttorneyUno"})
newq

Id StateAbbr                           QuestionUno  \
2            3        NC  BFC28E08-B6B1-4185-A91A-89DF90816BE4   
3            4        NC  45B24109-243A-489B-B0D2-2F36C2A369FE   
8            9        NC  459A9C10-12DF-4548-825D-EB7864457CC6   
9           10        NC  59D6CCAC-2EA5-4B62-8D38-17C4681DC856   
11          12        NC  119A97CE-60C3-412C-ADA1-5893D430BAE2   
...        ...       ...                                   ...   
202867  202868        NC  8CBF4481-97FF-476C-B9F8-077AF707BBA6   
202869  202870        NC  18992054-FD58-4D53-B37F-21F8DD9E069A   
202874  202875        NC  376FFF48-9068-4A8F-9316-2EB5A7E38001   
202875  202876        NC  98AA49B1-D423-4D12-9E70-A5BD6663DFE5   
202877  202878        NC  DC374484-7C6E-44A3-9DB7-89B8868518F1   

                                 CategoryUno                  Category  \
2       C8E1275A-6C42-42E4-9B7D-231BAC438FC9         Individual Rights   
3       C932D197-7F43-4BCF-BE0B-FE5F5DBCB9EF       Family and Children   
8       E6F4FD7C-FCE9-4763-AADD-0544158DE0BB                     Other   
9       C932D197-7F43-4BCF-BE0B-FE5F5DBCB9EF       Family and Children   
11      E6F4FD7C-FCE9-4763-AADD-0544158DE0BB                     Other   
...                                      ...                       ...   
202867  C932D197-7F43-4BCF-BE0B-FE5F5DBCB9EF       Family and Children   
202869  A56CA695-DA51-419F-87CF-228B360F2E9B  Housing and Homelessness   
202874  E6F4FD7C-FCE9-4763-AADD-0544158DE0BB                     Other   
202875  E6F4FD7C-FCE9-4763-AADD-0544158DE0BB                     Other   
202877  E6F4FD7C-FCE9-4763-AADD-0544158DE0BB                     Other   

                              SubcategoryUno  \
2       73A59926-BA0B-44EC-ABF3-F4C9181EEA83   
3       E9F87919-0CD5-4D9D-AAC6-3CBC49132F1D   
8       A911AEC6-4E0E-45B3-8807-024EC3C2EADE   
9       8C7CE01E-1EDE-42C4-8C99-0233E565855F   
11      81D88E4F-5B6A-4C9B-BE9A-C4AEAD0B1D7D   
...                                      ...   
202867  E9F87919-0CD5-4D9D-AAC6-3CBC49132F1D   
202869  F4BB9A8A-D699-48FD-A6E7-58FB95CC4021   
202874  E3257971-6837-4BAD-B00D-E4040F0C8B8A   
202875  E3257971-6837-4BAD-B00D-E4040F0C8B8A   
202877  26D3CE4B-3540-477E-BA03-78EF65769A12   

                                            Subcategory  \
2                           Civil/Constitutional Rights   
3       Family/Divorce/Custody/Visitation/Child Support   
8                                           Expungement   
9            Estate/Wills/Inheritance/Power of Attorney   
11                                      Personal Injury   
...                                                 ...   
202867  Family/Divorce/Custody/Visitation/Child Support   
202869             Housing/Eviction/Foreclosure/Tenants   
202874                                  Civil Procedure   
202875                                  Civil Procedure   
202877                                            Other   

                            AskedByClientUno           AskedOnUtc  \
2       DF636AEE-7D08-4F90-9A5C-0A2A726C6B59  2019-06-19 23:57:18   
3       C9C3FA40-0D60-49B9-97F1-63AFEFAFA484  2019-06-20 02:33:14   
8       3A361F5F-C49D-41AF-8BC4-BC211D14D6E1  2019-06-20 21:26:13   
9       11262CC7-711D-4387-86D7-975C18DBDB96  2019-06-21 17:32:08   
11      10C82829-7592-4D22-B6AE-D482DABF4074  2019-06-21 21:27:20   
...                                      ...                  ...   
202867  04C05471-0451-4D67-8F9F-6179C3F2D5CF  2019-06-16 19:19:33   
202869  61AA3759-8212-44E5-B6F5-16355978B7A7  2019-06-16 19:51:24   
202874  82BDD11F-2168-4A44-BBFC-BA9B0543F289  2019-06-17 20:37:01   
202875  FF2959DA-CCA4-415A-BA9B-2D51F860C08D  2019-06-18 13:06:00   
202877  A40F8543-330C-43A1-97AC-53C01FD4013E  2019-06-19 03:23:46   

                          TakenByAttorneyUno           TakenOnUtc  \
2       FA34142B-1575-4720-981C-2D28C3560137  2019-06-22 02:37:16   
3       FA34142B-1575-4720-981C-2D28C3560137  2019-07-27 18:04:08   
8       

In [ ]:
att = pd.merge(attorney, newq, on='AttorneyUno')
att
att.to_csv('/content/drive/Shareddrives/Data Fest 2023/data/attorney_cases.csv', index=False)

In [ ]:
#att.to_csv('/content/drive/Shareddrives/Data Fest 2023/data/attorney_cases.csv', index=False)

In [ ]:
g1 = att.groupby('AttorneyUno')['Category'].sum()
g1

AttorneyUno
000B021F-9FBF-4058-8ADC-D0A6232B6C04                             Housing and Homelessness
0023FF7C-934D-4974-87A9-B90BE52D4D5B    Family and ChildrenFamily and ChildrenFamily a...
003C4876-C682-4E94-B161-4BE4B5AD0FBD                                  Family and Children
00401285-58EE-4E39-9807-87A570C4B9ED    Family and ChildrenFamily and ChildrenFamily a...
00609EE0-87BE-496A-9847-A711F4569A25    Housing and HomelessnessFamily and ChildrenFam...
                                                              ...                        
FF9B1182-D622-4028-BA8F-5911D1E6D809                                  Family and Children
FFC66FAF-38C4-4520-88DE-19BBD7BF5F26    Work, Employment and UnemploymentHousing and H...
FFD551B4-3BB5-4C00-9714-032776657EC0    Family and ChildrenFamily and ChildrenFamily a...
FFDDF712-4C58-4DB5-A63E-99355C6448A0               Family and ChildrenFamily and Children
FFEADD85-31FF-42D4-A8B3-BEF4F180D8FD                         Consumer Financial Question

In [ ]:
att['count'] = 1

In [ ]:
att

Id_x StateAbbr_x                           AttorneyUno        City  \
0           1          NH  7A2C12BE-9F37-4C9D-A6ED-41B916820CE0     Concord   
1           1          NH  7A2C12BE-9F37-4C9D-A6ED-41B916820CE0     Concord   
2           1          NH  7A2C12BE-9F37-4C9D-A6ED-41B916820CE0     Concord   
3           6          LA  231AB5DE-6431-4BAE-BAC3-0E453BFD7062   Covington   
4           6          LA  231AB5DE-6431-4BAE-BAC3-0E453BFD7062   Covington   
...       ...         ...                                   ...         ...   
137782  11539          VA  2BD64F66-B14E-4119-9B2F-6B22D6BB98AC  ALEXANDRIA   
137783  11539          VA  2BD64F66-B14E-4119-9B2F-6B22D6BB98AC  ALEXANDRIA   
137784  11539          VA  2BD64F66-B14E-4119-9B2F-6B22D6BB98AC  ALEXANDRIA   
137785  11539          VA  2BD64F66-B14E-4119-9B2F-6B22D6BB98AC  ALEXANDRIA   
137786  11543          VA  FE8B3662-7C55-416A-B123-C3F9A9F3CAFA    Richmond   

                    County      StateName  PostalCode           CreatedUtc  \
0                 Cheshire  New Hampshire        3301  2019-02-14 16:21:52   
1                 Cheshire  New Hampshire        3301  2019-02-14 16:21:52   
2                 Cheshire  New Hampshire        3301  2019-02-14 16:21:52   
3       St. Tammany Parish      Louisiana       70433  2016-08-16 19:45:15   
4       St. Tammany Parish      Louisiana       70433  2016-08-16 19:45:15   
...                    ...            ...         ...                  ...   
137782     Alexandria city       Virginia       22314  2021-11-01 20:06:00   
137783     Alexandria city       Virginia       22314  2021-11-01 20:06:00   
137784     Alexandria city       Virginia       22314  2021-11-01 20:06:00   
137785     Alexandria city       Virginia       22314  2021-11-01 20:06:00   
137786       Richmond city       Virginia       23225  2021-11-15 23:27:59   

          Id_y StateAbbr_y  ...                           Category  \
0       160969          NH  ...                              Other   
1       161041          NH  ...                Family and Children   
2       161286          NH  ...                Family and Children   
3       155969          LA  ...                Family and Children   
4       155970          LA  ...                Family and Children   
...        ...         ...  ...                                ...   
137782   71375          VA  ...           Housing and Homelessness   
137783   71378          VA  ...                          Education   
137784   71407          VA  ...                              Other   
137785   71428          VA  ...  Work, Employment and Unemployment   
137786   71421          VA  ...                              Other   

                              SubcategoryUno                Subcategory  \
0       A36C230B-2CD3-40D8-A44D-6D1AAC88DDC0                      Other   
1       789FE53A-EC4F-4BAF-8F8E-7814D2AA8CFC     Family/Divorce/Custody   
2       789FE53A-EC4F-4BAF-8F8E-7814D2AA8CFC     Family/Divorce/Custody   
3       488FD95A-23A5-4313-ACBD-0A300F46608B     Family/Divorce/Custody   
4       488FD95A-23A5-4313-ACBD-0A300F46608B     Family/Divorce/Custody   
...                                      ...                        ...   
137782  216AA505-1C74-486D-BD3A-0DE059BBD9C0  Housing or Property Owned   
137783  5FAD8052-5F15-495E-A82A-147213A900B9          School Discipline   
137784  57F80FCA-C17B-42CF-A2BF-4D9E668C876E                      Other   
137785  CF72BA3E-76E5-4830-A54C-2C7E4F46FE5A                       Work   
137786  57F80FCA-C17B-42CF-A2BF-4D9E668C876E                      Other   

                            AskedByClientUno           AskedOnUtc  \
0       AD0F0934-9FFE-43CB-B01A-6A3450335442  2019-06-05 10:10:19   
1       79A7EFC7-A5D6-4F0F-AEC6-E30F9405BC72  2019-08-29 16:58:57   
2       1B6F52CF-304F-44EC-AF8B-A7B3BD9283F8  2020-05-30 11:02:49   
3       C876E564-7C7E-47A3-ADA4-65EA54D8C28A  2016-12-29 04:39:31   
4       6EA64518-E088-4D2C-82D4-9B01

In [ ]:
g2 = att.groupby(['AttorneyUno','Category'])['count'].sum()

attcatcount = pd.DataFrame(g2)
attcatcount


count
AttorneyUno                          Category                                
000B021F-9FBF-4058-8ADC-D0A6232B6C04 Housing and Homelessness               1
0023FF7C-934D-4974-87A9-B90BE52D4D5B Family and Children                    3
003C4876-C682-4E94-B161-4BE4B5AD0FBD Family and Children                    1
00401285-58EE-4E39-9807-87A570C4B9ED Family and Children                    3
00609EE0-87BE-496A-9847-A711F4569A25 Family and Children                    4
...                                                                       ...
FFC66FAF-38C4-4520-88DE-19BBD7BF5F26 Work, Employment and Unemployment      1
FFD551B4-3BB5-4C00-9714-032776657EC0 Family and Children                    4
                                     Work, Employment and Unemployment      1
FFDDF712-4C58-4DB5-A63E-99355C6448A0 Family and Children                    2
FFEADD85-31FF-42D4-A8B3-BEF4F180D8FD Consumer Financial Questions           1

[12294 rows x 1 columns]

In [ ]:
#attcatcount.to_csv('/content/drive/Shareddrives/Data Fest 2023/data/attorney_cat_count.csv', index=False)

In [ ]:
attcatcount

count
AttorneyUno                          Category                                
000B021F-9FBF-4058-8ADC-D0A6232B6C04 Housing and Homelessness               1
0023FF7C-934D-4974-87A9-B90BE52D4D5B Family and Children                    3
003C4876-C682-4E94-B161-4BE4B5AD0FBD Family and Children                    1
00401285-58EE-4E39-9807-87A570C4B9ED Family and Children                    3
00609EE0-87BE-496A-9847-A711F4569A25 Family and Children                    4
...                                                                       ...
FFC66FAF-38C4-4520-88DE-19BBD7BF5F26 Work, Employment and Unemployment      1
FFD551B4-3BB5-4C00-9714-032776657EC0 Family and Children                    4
                                     Work, Employment and Unemployment      1
FFDDF712-4C58-4DB5-A63E-99355C6448A0 Family and Children                    2
FFEADD85-31FF-42D4-A8B3-BEF4F180D8FD Consumer Financial Questions           1

[12294 rows x 1 columns]